emukit/quadrature/interfaces/standard_kernels.py

```diff
@@ -64,4 +64,40 @@ def dKdiag_dx(self, x: np.ndarray) -> np.ndarray:
         :param x: Argument of the kernel, shape = (n_points M, input_dim).
         :return: The gradient of the diagonal of the kernel evaluated at x, shape (input_dim, M).
         """
-        raise np.zeros((x.shape[1], x.shape[0]))
+        return np.zeros((x.shape[1], x.shape[0]))
+
+
+class IProductMatern32(IStandardKernel):
+    """Interface for a Matern32 product kernel.
+
+    Inherit from this class to wrap your ProductMatern32 kernel.
+
+    The ProductMatern32 kernel is of the form
+    :math:`k(x, x') = \sigma^2 \prod_{i=1}^d k_i(x, x')` where
+
+    .. math::
+        k_i(x, x') = (1 + \sqrt{3}r_i ) e^{-\sqrt{3} r_i}.
+
+    :math:`d` is the input dimensionality,
+    :math:`r_i:=\frac{|x_i - z_i|}{\lambda_i}}`,
+    :math:`\sigma^2` is the ``variance`` property and :math:`\lambda_i` is the ith element
+    of the ``lengthscales`` property.
+
+    Make sure to encode only a single variance parameter, and not one for each individual :math:`k_i`.
+    """
+
+    @property
+    def lengthscales(self) -> np.ndarray:
+        raise NotImplementedError
+
+    @property
+    def variance(self) -> np.float:
+        raise NotImplementedError
```

Comment:
Would be lovely to have docstrings here

Reply:
Yes, there are a few docstrings missing also in the existing kernel implementations. I may go trough all of them soon anyways, then I can address this.